### ISSUE TITLE
sort-imports should treat single import of multiple style as "multiple" type

### ISSUE THREAD

**Issue by LegNeato** [eslint/eslint#6736]
**What version of ESLint are you using?**
HEAD

**What rule do you want to change?**
`sort-imports`

**What code should no longer be flagged as incorrect with this change?**

``` js
import * as bar from 'bar.js';
import { a } from 'foo';
```

**What happens when the rule is applied to this code now?**
`X:Y  error  Imports should be sorted alphabetically`

**Root cause**
The rule uses `node.specifiers.length` to determine if an import is of type `multiple` or `single`. The bug is that imports of `multiple`-style with a single import (`import { a } from 'foo';`) are treated as `single`.

**Comment by LegNeato** [comment]
I have a patch for this.

**Comment by platinumazure** [comment]
Thanks @LegNeato! Seems like this could be a bug-- hopefully the team will accept pretty quickly. (I'm not familiar with import statements, so I don't feel comfortable accepting this myself.)

**Comment by btmills** [comment]
I'm not able to reproduce this. @LegNeato any ideas?

```js
// test6736.js
import * as bar from 'bar.js';
import { a } from 'foo';
```

```js
// config6736.js
module.exports = {
	"parserOptions": {
		"sourceType": "module"
	},

	"rules": {
		"sort-imports": "error"
	}
};
```

```sh
$ git log --oneline -n 1
71ae64c Docs: Clarify cache file deletion (fixes #4943) (#6712)

$ bin/eslint.js --no-eslintrc --config config6736.js test6736.js

$
```

**Comment by LegNeato** [comment]
@btmills I wonder if it is because I am using `babel-eslint` and it is doing something funky?

**Comment by btmills** [comment]
That's quite possible. Can you try it with the default parser?

**Comment by ilyavolodin** [comment]
@LegNeato Could you confirm that this is only an issue with `babel-eslint`?

**Comment by LegNeato** [comment]
@ilyavolodin I believe this **is an issue with the default parser** too. The reason why the above wasn't erroring is because it matches the default behavior--changing the example shows the bug (sorry for picking a bad example):

```js
// test6736.js
import bar from 'bar.js'; // Single
import { a } from 'foo'; // Multiple
```

```js
// config6736.js
module.exports = {
    "parserOptions": {
        "sourceType": "module"
    },

    "rules": {
        "sort-imports": ["error", {
        "memberSyntaxSortOrder": ["none", "single", "all", "multiple"]
        }]
    }
};
```

Note that `memberSyntaxSortOrder` has `single` before `multiple` (matching the code) so this should not produce an error. It incorrectly produces an error without my patch applied:

```sh
$ git log --oneline -n 1
f8ab8f1 New: func-call-spacing rule (fixes #6080) (#6749)

 ./bin/eslint.js  --no-eslintrc --config config6736.js test6736.js

/whatever/test6736.js
  4:1  error  Imports should be sorted alphabetically  sort-imports

```

Changing the example to have more than one "multiple" import (which should sort the same as the above example) produces different results:

```js
// test6736_many.js
import bar from 'bar.js'; // Single
import { a, b } from 'foo'; // Multiple
```

```sh
$ git log --oneline -n 1
f8ab8f1 New: func-call-spacing rule (fixes #6080) (#6749)

 ./bin/eslint.js  --no-eslintrc --config config6736.js test6736_many.js

$

```

**Comment by LegNeato** [comment]
@ilyavolodin 👈 friendly poke

**Comment by btmills** [comment]
@LegNeato thanks for your patience and for putting together that repro. I'm getting the same results now.

Before I consider the bug verified and accept the issue, are we comfortable treating the "single" option as a synonym for default imports and the "multiple" option as a synonym for named imports?

If accepted as stated, this would be a semver-minor change, as it could result in more errors in the following case:

```js
// test6736_reversed.js
import { a } from 'foo'; // Multiple
import bar from 'bar.js'; // Single
```

Currently, the above case does not error. After this change, it would warn.

**Comment by kaicataldo** [comment]
I agree with this change, but I also think the options should then be renamed (in a backwards-compatible way) to something like: `["none", "all", "named", "default"]`

**Comment by btmills** [comment]
It seems like the rule didn't really account for these sorts of cases initially. How should it handle `import foo, { bar } from 'module';`?

**Comment by kaicataldo** [comment]
Great point - seems like that would have to be another category...

**Comment by kaicataldo** [comment]
I'm closing this issue because it looks like consensus couldn't be reached. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. We've found that issues failing to reach consensus after 21 days tend to never reach consensus, and as a team have decided to close such issues. This doesn't mean the idea isn't interesting or valuable, only that it's not something the team can commit to at the moment.
